### PR TITLE
Issue #507 Implement cleanup feature for macos

### PR DESCRIPTION
**Pull request by praveenkumar**
**Fixes:** Issue #507 

## Solution/Idea

Implement the cleanup for macos so user can run `crc cleanup` and it removes the file which created by setup command including the tray.

## Proposed changes

List main as well as consequential changes you introduced or had to introduce.

1. Add `cleanup` command.

## Testing

Run `crc cleanup` command and it should remove the files and change the file permission back to original user which created by `crc setup` command.

### CODE REVIEW

**Comment by anjannath** [comment]
Not related to this PR but to cleanup feature, i think the cleanups should also do the checks, if we perform the checks in `doPreflightCleanUp` and skip over the ones that fails, we'll only do cleanup for the needed ones, now even when tray is not installed, it tries to unload and remove it.

**Comment by praveenkumar** [comment]
> Not related to this PR but to cleanup feature, i think the cleanups should also do the checks, if we perform the checks in doPreflightCleanUp and skip over the ones that fails, we'll only do cleanup for the needed ones, now even when tray is not installed, it tries to unload and remove it.

@anjannath that is by design, we (me and @cfergeau ) had same conversation when implementing it for linux, we don't care if something is present or not but our cleanup stuff should run for each steps even a skip one.

File: pkg/crc/preflight/preflight_darwin.go [hunk 1]
```diff
@@ -71,18 +77,22 @@ var traySetupChecks = [...]PreflightCheck{
...
-		check:            checkIfDaemonAgentRunning,
-		fixDescription:   "Starting CodeReady Containers daemon",
-		fix:              fixDaemonAgentRunning,
-		flags:            SetupOnly,
+		checkDescription:   "Checking if CodeReady Containers daemon is running",
+		check:              checkIfDaemonAgentRunning,
+		fixDescription:     "Starting CodeReady Containers daemon",
+		fix:                fixDaemonAgentRunning,
+		flags:              SetupOnly,
+		cleanupDescription: "Unload CodeReady Containers daemon",
```
Comment (cfergeau) [inline pkg/crc/preflight/preflight_darwin.go:85]: Unloading?

File: pkg/crc/preflight/preflight_darwin.go [hunk 2]
```diff
@@ -50,18 +52,22 @@ var traySetupChecks = [...]PreflightCheck{
...
-		check:            checkIfDaemonPlistFileExists,
-		fixDescription:   "Creating launchd configuration for daemon",
-		fix:              fixDaemonPlistFileExists,
-		flags:            SetupOnly,
+		checkDescription:   "Checking if launchd configuration for daemon exists",
+		check:              checkIfDaemonPlistFileExists,
+		fixDescription:     "Creating launchd configuration for daemon",
+		fix:                fixDaemonPlistFileExists,
+		flags:              SetupOnly,
+		cleanupDescription: "Removing launchd configuration for daemon",
```
Comment (cfergeau) [inline pkg/crc/preflight/preflight_darwin.go:60]: This is consistent with the other messages, but maybe they should all be changed to  `CodeReady Containers daemon`

File: pkg/crc/preflight/preflight_darwin.go [hunk 3]
```diff
@@ -117,6 +127,12 @@ func RegisterSettings() {
 }
 
 func CleanUpHost() {
-	logging.Warn("Cleanup is not supported for MacOS")
-	doCleanUpPreflightChecks(getPreflightChecks())
+	// A user can use setup with experiment flag
```
Comment (cfergeau) [inline pkg/crc/preflight/preflight_darwin.go:130]: the flag is named 'experimental'.
I guess we could name it in full `with the --enable-experimental-features` flag

File: pkg/crc/preflight/preflight_darwin.go [hunk 4]
```diff
@@ -117,6 +127,12 @@ func RegisterSettings() {
...
+	// and not use cleanup with same flag, to avoid
```
Comment (cfergeau) [inline pkg/crc/preflight/preflight_darwin.go:131]: `cleanup` does not have an 'experimental' flag, does it?

**Pull request merged**